src/nvim/eval.c

```diff
@@ -18148,45 +18149,147 @@ void free_tv(typval_T *varp)
   }
 }
 
-/*
- * Free the memory for a variable value and set the value to NULL or 0.
- */
+#define TYPVAL_ENCODE_ALLOW_SPECIALS false
+
+#define TYPVAL_ENCODE_CONV_NIL() \
+    do { \
+      tv->vval.v_special = kSpecialVarFalse; \
+      tv->v_lock = VAR_UNLOCKED; \
+    } while (0)
+
+#define TYPVAL_ENCODE_CONV_BOOL(ignored) \
+    TYPVAL_ENCODE_CONV_NIL()
+
+#define TYPVAL_ENCODE_CONV_NUMBER(ignored) \
+    do { \
+      (void)ignored; \
```

Comment:
@justinmk If you see me writing something like this it means I got compiler warning. I cannot recall details now, but I do not use such code without purpose. If I am not mistaking problem is code in the unreachable `if(TYPVAL_ENCODE_ALLOW_SPECIALS && …)`: even though it is unreachable, compiler still checks this before optimizing out.